lib/state/ui/actions.js

```diff
@@ -0,0 +1,26 @@
+import {
+	VISIBLE_PANES_ADD,
+	VISIBLE_PANES_REMOVE,
+	VISIBLE_PANES_SET,
+} from '../action-types';
+import { DEFAULT_VISIBLE_PANES } from './constants';
+
+export const enterDistractionFree = () => ( {
+	type: VISIBLE_PANES_SET,
+	panes: [ 'editor' ],
+} );
+
+export const leaveDistractionFree = ( previousPanes = DEFAULT_VISIBLE_PANES ) => ( {
+	type: VISIBLE_PANES_SET,
+	panes: previousPanes,
+} );
```

Comment:
For now why don't we limit this to existing behaviors?

Reply:
👍 
You're right, I was experimenting on this and then decided to drop it and forgot to remove here.

Reply:
We do not currently have a distraction free mode though.

If we want to toggle between "editor + panes" / "just editor", we need to know which panes were visible before hiding them.

See for example @drw158 mockup:
- `[ notes, editor ]` -> enter DF -> `[ editor ]` -> leave DF -> `[ notes, editor ]`
- `[ tags, notes, editor ]` -> enter DF -> `[ editor ]` -> leave DF -> `[ tags, notes, editor ]`

My idea is to store the `previousPanes` list into the component internal state, and dispatch it accordingly.
If we don't dispatch a `previousPanes`, no harm done, we just return to the default state.